layers/water-mesh.js

```diff
@@ -200,10 +235,130 @@ export class WaterMesh extends BufferedMesh {
         this.geometryBindings.delete(key);
       }
     }
+    {
+      const physicsObject = this.physicsObjectsMap.get(key);
+
+      if (physicsObject) {
+        this.physics.removeGeometry(physicsObject);
+        this.physicsObjectsMap.delete(key);
+      }
+    }
+    {
+      const chunkKey = getHashKey(chunk.min.x, chunk.min.y);
+      this.chunkPhysicObjcetMap.delete(chunkKey);
+      this.waterHeightMap.delete(chunkKey);
+    }
     {
       const task = this.gpuTasks.get(key);
       task.cancel();
       this.gpuTasks.delete(key);
     }
   }
+
+  checkWaterContact(chunkPhysicObject, player, waterSurfaceHeight) {
+    // use overlapBox to check whether player contact the water
+    this.physics.enableGeometryQueries(chunkPhysicObject);
+    if (player.avatar) {
+      let collisionIds;
+      const height = player.avatar.height;
+      const width = player.avatar.shoulderWidth
+      if (player.position.y > waterSurfaceHeight) {
+        collisionIds = this.physics.overlapBox(width, height, width, player.position, player.quaternion).objectIds;
+      }
+      else {
+        localVector.set(player.position.x, waterSurfaceHeight, player.position.z);
+        collisionIds = this.physics.overlapBox(width, height, width, localVector, player.quaternion).objectIds;
+      } 
+      for (const collisionId of collisionIds) {
+        if (collisionId === chunkPhysicObject.physicsId) {
+          this.physics.disableGeometryQueries(chunkPhysicObject);

```

Comment:
Use functions to better cleanup this code
comments are fine too, but functions with good name are better imo

Reply:
[add comment](https://github.com/webaverse/heightfield/pull/6/commits/4509f2375f35e717d470853188c325f3653d79b2)

Reply:
[add comment](https://github.com/webaverse/heightfield/pull/6/commits/b55e6495c64970da5cebf6484782c051df2d4982)